File: web/src/types/collection.ts
/** Target collection domain types — mirror the Go backend domain model. */

export interface Collection {
  id: string
  name: string
  creatorId: string
  teamId: string
  targetIds: string[]
  createdAt: string
  updatedAt: string
}

export interface CreateCollectionInput {
  name: string
  targetIds: string[]
}

export interface UpdateCollectionInput {
  id: string
  name: string
  targetIds: string[]
}

File: internal/domain/collection.go
package domain

import "time"

// Collection represents a user-created saved group of targets.
type Collection struct {
	ID        string    `json:"id"`
	Name      string    `json:"name"`
	CreatorID string    `json:"creatorId"`
	TeamID    string    `json:"teamId"`
	TargetIDs []string  `json:"targetIds"`
	CreatedAt time.Time `json:"createdAt"`
	UpdatedAt time.Time `json:"updatedAt"`
}
